src/site/stages/build/plugins/add-nonce-to-scripts.js

```diff
@@ -1,18 +1,25 @@
 /* eslint-disable no-param-reassign */
 const path = require('path');
+const crypto = require('crypto');
 
 const CSP_NONCE = '**CSP_NONCE**';
 
-function generateNewId(existingIds) {
-  const newId = Math.random()
-    .toString(36)
-    .replace(/[^a-z]+/g, '')
-    .substr(0, 10);
-  if (!existingIds.has(newId)) {
-    existingIds.add(newId);
-    return newId;
-  }
-  return generateNewId(existingIds);
+function idGeneratorFactory(fileName) {
+  let i = 0;
+  const existingIds = new Set();
+
+  return function idGenerator() {
+    const newId = crypto
+      .createHash('md5')
+      .update(fileName + i)
+      .digest('hex');
+    i++;
+    if (!existingIds.has(newId)) {
+      existingIds.add(newId);
+      return newId;
+    }
+    return idGenerator();
```

Comment:
Is it really necessary to do this recursively? How often do we have ID conflicts?

Reply:
Prolly never, but my paranoia didn't let me _not_ keep the safety check. :stuck_out_tongue:

Reply:
Collisions would only happen when `md5(fileName + i) == md5(fileName + j)`, where `maxAddedIds >= j > i`. Since we're only doing that on maybe two or three in a file at most, we're likely never going to need the safety check, but it was already there from the previous implementation, so I went ahead and kept it. :man_shrugging:

Reply:
When I see a strange condition like that that seems like should never happen, I usually just throw an error there, then run a few builds to see if it ever happens. If it doesn't ever hit, I'll just remove the condition because it's just unnecessary complexity (which we have a lot of!) Only saying this because that's what I ran into yesterday with this strange condition - https://github.com/department-of-veterans-affairs/vets-website/pull/10837/files#diff-f171b9525a3be6fa8f25af23107e1bdfL21